index.d.ts

```diff
@@ -0,0 +1,130 @@
+declare module 'fluture' {
+  interface ResolveFunction<T> {
+    (value: T): void
+  }
+  interface RejectFunction {
+    (error: Error): void
+  }
+
+  interface FunctionReturning<T extends Function> {
+    (...t: any[]): T
+  }
+  abstract class Future<T> {
```

Comment:
Can we define `Future<L, R>` instead, in order to define restrictions for the "error" branch?